src/processor/operator/simple/export_db.cpp

```diff
@@ -124,23 +124,73 @@ static void exportLoadedExtensions(stringstream& ss, const ClientContext* client
     }
 }
 
-std::string getSchemaCypher(ClientContext* clientContext) {
+enum class EntryType { NODE, REL, SEQUENCE, UNREACHABLE };
+
+struct EntryAndType {
+    CatalogEntry* entry = nullptr;
+    EntryType type = EntryType::UNREACHABLE;
+};
+
+std::string getSchemaCypher(ClientContext* clientContext, bool sortInternalIds) {
     stringstream ss;
     exportLoadedExtensions(ss, clientContext);
     const auto catalog = clientContext->getCatalog();
     auto transaction = clientContext->getTransaction();
+
+    auto nodeTableEntries = catalog->getNodeTableEntries(transaction, false);
+    auto relGroupEntries = catalog->getRelGroupEntries(transaction, false);
+    auto sequenceEntries = catalog->getSequenceEntries(transaction);
     ToCypherInfo toCypherInfo;
-    for (const auto& nodeTableEntry :
-        catalog->getNodeTableEntries(transaction, false /* useInternal */)) {
-        ss << nodeTableEntry->toCypher(toCypherInfo) << std::endl;
-    }
     RelGroupToCypherInfo relTableToCypherInfo{clientContext};
-    for (const auto& entry : catalog->getRelGroupEntries(transaction, false /* useInternal */)) {
-        ss << entry->toCypher(relTableToCypherInfo) << std::endl;
-    }
     RelGroupToCypherInfo relGroupToCypherInfo{clientContext};
-    for (const auto sequenceEntry : catalog->getSequenceEntries(transaction)) {
-        ss << sequenceEntry->toCypher(relGroupToCypherInfo) << std::endl;
+
+    if (sortInternalIds) {
+        std::vector<EntryAndType> allEntries;
+        for (auto* e : nodeTableEntries) {
+            allEntries.push_back({static_cast<CatalogEntry*>(e), EntryType::NODE});
+        }
+        for (auto* e : relGroupEntries) {
+            allEntries.push_back({static_cast<CatalogEntry*>(e), EntryType::REL});
+        }
+        for (auto* e : sequenceEntries) {
+            allEntries.push_back({static_cast<CatalogEntry*>(e), EntryType::SEQUENCE});
+        }
+        std::sort(allEntries.begin(), allEntries.end(),
+            [](const EntryAndType& a, const EntryAndType& b) {
+                return a.entry->getOID() < b.entry->getOID();
+            });
+        for (const auto& entryWithType : allEntries) {
```

Comment:
The outputing logics are quite similar with/without sorting. Consider refactoring them.